mlflow/types/type_hints.py

```diff
@@ -0,0 +1,297 @@
+import inspect
+import logging
+from datetime import datetime
+from typing import Any, Literal, NamedTuple, Optional, Union, get_args, get_origin
+
+import pydantic
+
+from mlflow.exceptions import MlflowException
+from mlflow.types.schema import (
+    COLSPEC_TYPES,
+    AnyType,
+    Array,
+    ColSpec,
+    DataType,
+    Map,
+    Object,
+    Property,
+    Schema,
+)
+
+PYDANTIC_V2 = pydantic.VERSION.startswith("2.")
```

Comment:
What should happen if pydantic 3.0 is released?

Reply:
Can we pin pydantic < 2 in pyproject.toml file?

Reply:
I'd use PYDANTIC_V1 instead.